scripts/clang-tidy-only-diff.sh

```diff
@@ -41,20 +41,25 @@ if [ ${#modified_filepaths[@]} = 0 ]; then
   exit 0
 fi
 
+readonly num_jobs="${1:-$(nproc)}"
 # -m specifies that `parallel` should distribute the arguments evenly across the executing jobs.
 # -p Tells clang-tidy where to find the `compile_commands.json`.
 # `{}` specifies where `parallel` adds the command-line arguments.
 # `:::` separates the command `parallel` should execute from the arguments it should pass to the commands.
-parallel -m \
+parallel \
+  -m \
+  -j $num_jobs \
+  --halt-on-error now,fail=1 \
   ${CLANG_TIDY} \
   -p $tidy_dir \
   -header-filter "^$(pwd)/valhalla/[^/]+$" \
   -fix \
+  -fix-errors \
   -format-style=file \
-  {} ::: "${modified_filepaths[@]}"
+  {} ::: \
+  "${modified_filepaths[@]}"
 
-# TODO Enable exit-code 1 on errors once we fix all the existing clang-tidy errors to prevent noisy PR's
-#if ! git diff --exit-code --quiet; then
-#  echo "Tidy introduced changes"
-#  exit 1
-#fi
+if ! git diff --exit-code; then
+  echo "Tidy introduced changes"
```

Comment:
Optional: what do you think about printing out the diff here?

Reply:
100% i think that is what we should do. print it out for our benefit but not fail the build

Reply:
It actually outputs the diff one line above, line 62.

Reply:
My vote: print out the diff  *and* fail the build (so that folks actually look at it).